
<launch>
 <!-- ***************************** 
 needed :

 - map_server ==> load map
 - amcl ==> localization
 - wheel speed ==> odom
 - pointcloud_to_laserscan ==> scan
 - aibee_navi ==> global planner & local planner
 - cmd_control == > cmd or ipad control change
 - obstable_navigatoin ==>
 ************************************ -->


<!-- ***************************** ROS Navigation ************************************ -->
<param name="/use_sim_time" value="true" />
<!-- aibee_robot urdf -->
<include file="$(find aibee_robot)/launch/aibee_robot_urdf.launch" />

<!--map server 提供 /map -->
<arg name="map" default="office_map/map_office.yaml" />
<node name="map_server" pkg="map_server" type="map_server" args="$(find aibee_robot)/maps/$(arg map)">
<param name="map_topic" value="map"/> 
</node>

<!--wheel speed  提供/odom-->
<include file="$(find wheel_speed_odo)/launch/odom_use_simtime.launch"/> 

<!-- amcl -->
<include file="$(find amcl)/launch/amcl_diff.launch"/> 



<!--pointCloud to Laserscan 提供/scan  vertical and horizental-->
<include file="$(find pointcloud_to_laserscan)/launch/vscan_horizental.launch"/>
<include file="$(find pointcloud_to_laserscan)/launch/vscan_vertical.launch"/> 

<!-- sonars to poincloud node  -->
<node name="sonars_to_pointcloud_node" pkg="sonars_to_pointcloud" type="sonars_to_pointcloud" output="screen">
</node>

<!-- aibee_robot navigation action -->
  <node pkg="aibee_robot" type="aibee_navi_node"  name="aibee_navi" output="screen" >
    <param name="map_topic" value="map"/>  
  
    <rosparam file="$(find aibee_robot)/config/global_planner_params.yaml" command="load" />
  
    <rosparam file="$(find aibee_robot)/config/local_planner_params.yaml" command="load" />
  </node>
  

<!-- cmd or ipad control change-->
<node pkg="cmd_control" type="cmd_control_node"  name="cmd_control_node" output="screen"/>

  <!-- node function: 订阅ipad点位, 利用movebase 导航到对应位置-->
  <node name="obstable_navigatoin" pkg="obstable_navigatoin" type="obstable_navigatoin_node" output="screen" >
  </node> 

<node pkg="cmd_control" type="auto_move_node"  name="auto_move_node" output="screen" >
  <param name="waypoints_num" value="5"/>
</node>

 <!-- node pkg="rviz" type="rviz" name="rviz" args="-d $(find aibee_robot)/rviz/obstable.rviz"/ -->

</launch>